stock_account_product_run_fifo_hook/hooks.py

```diff
@@ -0,0 +1,348 @@
+                    current_real_time_svls |= svl_to_vacuum
+            real_time_svls_to_vacuum |= current_real_time_svls
+        ValuationLayer.sudo().create(new_svl_vals_manual)
+        vacuum_svls = ValuationLayer.sudo().create(new_svl_vals_real_time)
+
+        # If some negative stock were fixed, we need to recompute the standard price.
+        for product in self:
+            product = product.with_company(company.id)
+            if product.cost_method == "average" and not float_is_zero(
+                product.quantity_svl, precision_rounding=product.uom_id.rounding
+            ):
+                product.sudo().with_context(disable_auto_svl=True).write(
+                    {"standard_price": product.value_svl / product.quantity_svl}
+                )
+
+        vacuum_svls._validate_accounting_entries()
+        self._create_fifo_vacuum_anglo_saxon_expense_entry(
+            vacuum_svls, real_time_svls_to_vacuum
+        )
+
+    if not hasattr(ProductProduct, "_run_fifo_vacuum_original"):
+        ProductProduct._run_fifo_vacuum_original = ProductProduct._run_fifo_vacuum
+    ProductProduct._run_fifo_vacuum = _run_fifo_vacuum_new
+
+    def _create_out_svl_new(self, forced_quantity=None):
+        """Create a `stock.valuation.layer` from `self`.
+
+        :param forced_quantity: under some circumstances,
+        the quantity to value is different than
+            the initial demand of the move (Default value = None)
+        """
+        svl_vals_list = []
+        for move in self:
+            move = move.with_company(move.company_id)
+            valued_move_lines = move._get_out_move_lines()
+            valued_quantity = 0
+            for valued_move_line in valued_move_lines:
+                valued_quantity += valued_move_line.product_uom_id._compute_quantity(
+                    valued_move_line.quantity, move.product_id.uom_id
+                )
+            if float_is_zero(
+                forced_quantity or valued_quantity,
+                precision_rounding=move.product_id.uom_id.rounding,
+            ):
+                continue
+            svl_vals = move.product_id.with_context(
+                move_requestor=move.id
+            )._prepare_out_svl_vals(forced_quantity or valued_quantity, move.company_id)
+            svl_vals.update(move._prepare_common_svl_vals())
+            if forced_quantity:
+                svl_vals["description"] = (
+                    _("Correction of %s (modification of past move)")
+                    % move.picking_id.name
+                    or move.name
+                )
+            svl_vals["description"] += svl_vals.pop("rounding_adjustment", "")
+            svl_vals_list.append(svl_vals)
+        return self.env["stock.valuation.layer"].sudo().create(svl_vals_list)
+
+    if not hasattr(StockMove, "_create_out_svl_original"):
+        StockMove._create_out_svl_original = StockMove._create_out_svl
+    StockMove._create_out_svl = _create_out_svl_new
```

Comment:
you will have to update method, due to https://github.com/odoo/odoo/commit/c8f808a8dff10b9043e23cf105c828f56807dbbb